src/main/java/online/book/store/controller/AuthController.java

```diff
@@ -0,0 +1,30 @@
+package online.book.store.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import online.book.store.dto.user.request.UserRegistrationRequestDto;
+import online.book.store.dto.user.response.UserResponseDto;
+import online.book.store.exception.RegistrationException;
+import online.book.store.service.UserService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "User Authorization Management",
+        description = "Endpoints for managing user authorization")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/auth")
+public class AuthController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    @Operation(summary = "Register a new user", description = "Register a new user")
+    public UserResponseDto registerUser(
+            @RequestParam UserRegistrationRequestDto userRequestDto)
```

Comment:
```suggestion
            @RequestBody UserRegistrationRequestDto userRequestDto)
```
Read about difference between RequestParam, PathVariable and RequestBody annotations